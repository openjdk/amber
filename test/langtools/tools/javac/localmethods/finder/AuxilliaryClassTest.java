/**
 * @test /nodynamiccopyright/
 * @bug 8233754
 * @compile/fail/ref=AuxilliaryClassTest.out -Werror -XDrawDiagnostics -XDfind=local-methods AuxilliaryClassTest.java
 */

class AuxilliaryClassTest {

    private static void run() {
        run();
        run();
        run();
        run();
    } // can be local since recursion does not count.

    public static void main(String [] args) {
        run();
        run();
        run();
        run();  // multiple call sites from the same method is OK.
    }
}

class AuxilliaryClassTest01 {

    private static void run() {
        run();
        run();
        run();
        run();
    } // can be local since recursion does not count.

    public static void main(String [] args) {
        run();
        run();
        run();
        run();  // multiple call sites from the same method is OK.
    }
}
