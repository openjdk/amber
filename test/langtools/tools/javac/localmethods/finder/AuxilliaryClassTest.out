AuxilliaryClassTest.java:9:25: compiler.warn.method.could.be.declared.local: main(java.lang.String[]), kindname.class, AuxilliaryClassTest
- compiler.err.warnings.and.werror
AuxilliaryClassTest.java:26:25: compiler.warn.method.could.be.declared.local: main(java.lang.String[]), kindname.class, AuxilliaryClassTest01
1 error
2 warnings
